go/enclave/l2chain/interfaces.go

```diff
@@ -0,0 +1,32 @@
+package l2chain
+
+import (
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	gethcore "github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/vm"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+	"github.com/obscuronet/go-obscuro/go/common/gethapi"
+	"github.com/obscuronet/go-obscuro/go/enclave/core"
+)
+
+// ObscuroChain - the interface that provides the data access layer to the obscuro l2.
+// Operations here should be read only.
+type ObscuroChain interface {
+	// GetBalance - Returns the balance of an address given the standard provided BlockNumber which in our case is batch number.
+	// Note this method also returns the address we should encrypt the balance for, unlike `GetBalanceAtBlock`
```

Comment:
As far as I can tell, it's because the GetBalance is a User Method and GetBalanceAtBlock isn't.

These names are confusing. The GetBalance is a "special" function that does logic to figure out if the address is a contract or EOA so it can decide which keys it should encrypt with. 
The GetBalance calls the GetBalanceAtBlock to actually get the balance of the address.

Should the methods' names be distinguished ? One JUST fetches the balance and the other actually implements the rules required to safely return the balance to the user.